xdsl/dialects/gpu.py

```diff
@@ -405,13 +403,13 @@ def __init__(
         properties: dict[str, Attribute | None] = {
             "function_type": function_type,
         }
-        if knwown_block_size is not None:
-            attributes["gpu.known_block_size"] = ArrayAttr(
-                IntegerAttr(i, i32) for i in knwown_block_size
+        if known_block_size is not None:
+            attributes["gpu.known_block_size"] = DenseArrayBase.create_dense_int(
+                i32, known_block_size
             )
-        if knwown_grid_size is not None:
-            attributes["gpu.known_grid_size"] = ArrayAttr(
-                IntegerAttr(i, i32) for i in knwown_grid_size
+        if known_grid_size is not None:
+            attributes["gpu.known_grid_size"] = DenseArrayBase.create_dense_int(
+                i32, known_grid_size
```

Comment:
drive-by fix, this is a good reminder to avoid this pattern of building attributes up, as it erases all type information